test/e2e/space_autocompletion_test.go

```diff
@@ -0,0 +1,157 @@
+			// spaceIsPendingCluster(t, hostAwait, space2.Name)
+			//
+			//t.Run("reset the max number and expect the second space will be provisioned as well", func(t *testing.T) {
+			//	// when
+			//	hostAwait.UpdateToolchainConfig(testconfig.AutomaticApproval().MaxNumberOfUsers(1000))
+			//
+			//	// then
+			//	VerifyResourcesProvisionedForSpace(t, awaitilities, space2)
+			//})
+		})
+	})
+
+	t.Run("mark the first member cluster as full and for the second keep some capacity - expect that the space will be provisioned to the second one", func(t *testing.T) {
+		// given
+		var memberLimits []testconfig.PerMemberClusterOptionInt
+		toolchainStatus, err := hostAwait.WaitForToolchainStatus(wait.UntilToolchainStatusHasConditions(ToolchainStatusReadyAndUnreadyNotificationNotCreated()...))
+		require.NoError(t, err)
+		for _, m := range toolchainStatus.Status.Members {
+			if memberAwait1.ClusterName == m.ClusterName {
+				memberLimits = append(memberLimits, testconfig.PerMemberCluster(memberAwait1.ClusterName, m.UserAccountCount))
+			} else if memberAwait2.ClusterName == m.ClusterName {
+				memberLimits = append(memberLimits, testconfig.PerMemberCluster(memberAwait2.ClusterName, m.UserAccountCount+1))
+			}
+		}
+		require.Len(t, memberLimits, 2)
+
+		hostAwait.UpdateToolchainConfig(testconfig.AutomaticApproval().MaxNumberOfUsers(0, memberLimits...))
+
+		// when
+		space1 := CreateSpace(t, awaitilities, WithName("space-multimember-1"))
+
+		// then
+		VerifyResourcesProvisionedForSpace(t, awaitilities, space1, wait.UntilSpaceHasStatusTargetCluster(memberAwait2.ClusterName))
+
+		t.Run("after both members marking as full then the new space won't be provisioned", func(t *testing.T) {
+			// given
+			var memberLimits []testconfig.PerMemberClusterOptionInt
+			for _, m := range toolchainStatus.Status.Members {
+				if memberAwait1.ClusterName == m.ClusterName {
+					memberLimits = append(memberLimits, testconfig.PerMemberCluster(memberAwait1.ClusterName, m.UserAccountCount))
+				} else if memberAwait2.ClusterName == m.ClusterName {
+					memberLimits = append(memberLimits, testconfig.PerMemberCluster(memberAwait2.ClusterName, m.UserAccountCount))
+				}
+			}
+			require.Len(t, memberLimits, 2)
+			hostAwait.UpdateToolchainConfig(testconfig.AutomaticApproval().MaxNumberOfUsers(0, memberLimits...))
+
+			// when
+			space2 := CreateSpace(t, awaitilities, WithName("space-multimember-2"))
+
+			// then
+			spaceIsPendingCluster(t, hostAwait, space2.Name)
+
+			t.Run("when target cluster is set manually, then the limits will be ignored", func(t *testing.T) {
+				// when & then
+				CreateAndVerifySpace(t, awaitilities, WithName("space-multimember-3"), WithTargetCluster(memberAwait1))
+				// and still
+				spaceIsPendingCluster(t, hostAwait, space2.Name)
+			})
+		})
+	})
+}
+
+func spaceIsPendingCluster(t *testing.T, hostAwait *wait.HostAwaitility, name string) *toolchainv1alpha1.Space {
```

Comment:
could be renamed to `waitUntilSpaceIsPendingCluster`, for better readability/understandability

Reply:
yeah, why not: 0f4200f